### ISSUE TITLE
Mi Band 2 - FW 1.0.1.54 (recommended) does NOT connect! - Downgrade doesn't work!

### ISSUE THREAD

**Issue by momo42** [Freeyourgadget/Gadgetbridge#859]
#### Before opening an issue please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
I installed the recommended FW 1.0.1.54 (https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Information). I also checked the MD5SUM of the file before installation (https://github.com/Freeyourgadget/Gadgetbridge/wiki/Mi-Band-Firmware-Update).

After the Installation my Mi Band 2 didn't connect to my Phone. I had to delete and and add the Mi Band 2 again as described here: https://github.com/Freeyourgadget/Gadgetbridge/issues/847

~~This worked, but if I switch BT off and on (or restart the phone) the Mi Band 2 doesn't connect again.~~
So I have to repeat the delete-add-procedure over and over again.
edit: after some delete add cycles the Mi Band 2 connects again but auto connect still doesn't work

Two other things are suspicious:

1) The re added Mi Band 2 has the "HPlus/Makibes" icon in the Gadgetbridge main menu now.
2) And I'm not able to install Mili_pro.ft or Mili_pro.ft.en Files via FW/App Installer. The (german) error message says "Datei kann nicht installiert werden, Gerät nicht bereit."
edit: firmware installation also doesn't work.

I have this issue with two Bands and two (identical) phones.

#### Your wearable device is:
Mi Band 2 - FW 1.0.1.54

#### Your android version is:
Android 7.1.2 (LineageOS 14.1)

#### Your Gadgetbridge version is:
0.22.0

**Comment by momo42** [comment]
Ok, I tried to install an older recommended Firmware (1.0.1.53) but I also get the same error message:  "Datei kann nicht installiert werden, Gerät nicht bereit."

What can I do now? Please help!

**Comment by HennesTheSecond** [comment]
I can confirm this exact behaviour. Same FW, same Android version. I installed the FW a few days ago and had to delete and add the Mi Band 2. After a phone reboot it doesn't connect again. FW Installer doesn't work too. :-(

**Comment by momo42** [comment]
I deleted and re added the Mi Bands on both phones two or three times and now the Mi Bands are able to connect again after reboot. But you have to connect them manually (open up Gadgetbridge and click on the entry). In Gadgetbridge Options "Automatisch starten", "Verbinde, wenn Bluetooth engeschaltet" and "Verbindung automatisch wiederherstellen" are checked.

FW installer still doesn't work. So I still cannot downgrade.

**Comment by ashimokawa** [comment]
What I had to do after upgrading to .54 is to unpair the device in android settings and then press the + button in the main activity to pair again (no need to delete anything). This was a one time operation.

For the Amazfit Bip I had the same when updating to 0.0.9.14 (FW from the same Mifit 3.1.0 as .54 for Mi Band 2). The next update for the Bip did not have the problem after updating to I am quite confident that the next version for Mi Band 2 also wont have that problem.

Anyway I will remove and re-add the device to look for the FW Installer bug.

**Comment by ashimokawa** [comment]
I don't think there is any need to downgrade your firmware, just update to Gadgetbridge 0.22.1 (when it shows up on F-Droid, I just tagged it). After that you should unpair in Android settings again and use the + button in Gadgetbridge to pair again. Make sure the device is not connected during that operation. I never had to reboot or delete devices to make it work again.

Autoconnect also seems to work.

**Comment by ashimokawa** [comment]
@momo42 
@HennesTheSecond 
0.22.1 is not in F-Droid, please update and follow the instructions above to fix your situation

**Comment by momo42** [comment]
Thanks a lot!

0.22.1 recognizes our Mi Bands 2 correct again (with Mi Band icon in menu). And I was able to install the Fonts via Firmware Installer. I have to say that I deleted Gadgetbridge before and installed 0.22.1 fresh from F-Droid.

But one problem persists: The Mi Band 2 doesn't reconnect if phone was out of range or if phone gets restarted. Same behavior with two Phones (Android 7.1.2) and two Mi Band 2 (FW 1.0.1.54).

Unfortunately I don't remember my old Versions of Gadgetbridge and Mi Band Firmware but I'm sure this worked before. (All reconnect options are ticked in Gadgetbridge)

**Comment by HennesTheSecond** [comment]
Same here.
My Mi Band 2 has the right menu icon again but it does not auto connect anymore if the connection was lost (phone reboot or phone left in another room). This is absolutely a new behavior and very annoying if you rely on Mi Band vibration for message signaling. :-/

Most time, when I look on my phone, it says "not connected". Should I open a new Issue for that?

And another minor bug: FW Installer works now but "FW 1.0.1.54" is not shown as compatible FW version.

**Comment by ashimokawa** [comment]
@HennesTheSecond 
Rebooting the phone did not reconnect also for me.

When going out of range and then back in, reconnect works for me sometimes, sometimes not. Just tried it. 

Are you sure that this is due to a Gadgetbridge regression, or could it be the firmware?

**Comment by HennesTheSecond** [comment]
Hard to say. I'm sure that auto reconnect worked reproducible some months ago but unfortunately I upgraded Mi Band 2 firmware, Gadgetbridge AND Android in the meantime. Looks like a LOT of testing is required to locate the trigger for this regression. I also noticed that sometimes (rare) reconnect works without user interaction.

Unfortunately I'm in my exam period now. So I don't have much time to tinker around with my mobile gadgets. :-/

Would it help to open a new (open) issue for that auto reconnect problem? Perhaps other users have more time to investigate.
